tutorial/index.html

```diff
@@ -0,0 +1,412 @@
+        demand received from its Subscriber(s).</em></blockquote>
+
+    <p>
+        Akka implements the Reactive Streams SPI and fortunately the code to turn an actor into a Publisher is pretty
+        straightforward:
+        we just have to extend <code>ActorPublisher</code> which exposes (among others) the <code>onNext</code> which we
+        can use to push a new tweet to our subscribers when one arrives. Only thing left is to subscribe to the event
+        bus when the
+        actor is started. The code can be found in <a
+            href="#code/src/main/scala/reactive/tweets/outgoing/TweetPublisher.scala" class="shortcut">TweetPublisher.scala</a>
+        and should look like this:
+    </p>
+
+<pre><code>class TweetPublisher extends ActorPublisher[Tweet] {
+
+  override def preStart = {
+    context.system.eventStream.subscribe(self, classOf[Tweet])
+  }
+
+  override def receive = {
+    case tweet: Tweet => onNext(tweet)
+  }
+
+}</code></pre>
+
+    <p>
+        Next: <a href="#tutorial/5" class="shortcut">Running the Flow</a>
+    </p>
+</div>
+<div>
+    <h2>Part 1: Running the Flow</h2>
+
+    <p>
+        In the previous section we described how to define an actor as a Reactive Streams Publisher. To start streaming
+        we still need to instantiate the actor and - in one go - transform it into a Source. This is done in <a
+            href="#code/src/main/scala/reactive/tweets/outgoing/TweetFlow.scala" class="shortcut">TweetFlow.scala</a>.
+
+    <pre><code>private val tweetSource: Source[Tweet, ActorRef] =
+        Source.actorPublisher[Tweet](TweetPublisher.props)</code></pre>
+
+    Now we've got our hands on a proper <code>Source</code> we can use it to construct a <code>Flow</code>:
+
+        <pre><code>def tweetFlow(tweetFilter: TweetFilter): Flow[Message, Message, Unit] =
+            Flow.wrap(Sink.ignore, tweetSource filter tweetFilter map toMessage)(Keep.left)</code></pre>
+
+    Note: ignore the tweetFilter for now, this is just a convenience parameter to be able to construct specific streams,
+    like the stream of tweets for a specific user, or the stream of tweets that contain a specific hash tag.
+    </p>
+
+    <p>
+        We are only interested in one way communication with our Websocket. We push new tweets to the client on their
+        arrival. Theoretically we could also receive messages over the same channel from the client. We just ignore
+        those.
+        So what does this look like in terms of a Flow?
+    </p>
+
+    <pre>
+        +-----------------------------+
+        | Our Websocket Flow          |
+        |                             |
+        |  +--------+      +------+   |
+        |   \       |      |       \  |
+    In ~~>   | SINK |      | SOURCE |~~> Out
+        |   /       |      |       /  |
+        |  +--------+      +------+   |
+        +-----------------------------+
```

Comment:
You had such nice pictures earlier, and now you resort to ASCII art? :stuck_out_tongue_winking_eye: 
(Don't get me wrong, I love ASCII art. I wouldn't mind if you replaced all the pictures with it!)